index.js

```diff
@@ -46,21 +30,29 @@ module.exports._reset = function () {
  */
 module.exports.releaseExit = function() {
   if (exit) {
-    process.exit = exit;
+    _process.exit = exit;
     exit = null;
   }
 };
 
 var firstExitCode;
-
-module.exports.captureExit = function() {
+module.exports.captureExit = function(process) {
   if (exit) {
     // already captured, no need to do more work
     return;
   }
-  exit = process.exit;
 
-  process.exit = function(code) {
+  if (process instanceof EventEmitter === false) {
+    throw new TypeError('attempt to capture a bad process instance');
+  }
+
+  _process = process;
```

Comment:
I really don't like us using module state for this, but fixing that will be a refactoring we can add later...